secret-manager/runtime/src/main/java/io/quarkiverse/googlecloudservices/secretmanager/runtime/config/SecretManagerClientProvider.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkiverse.googlecloudservices.secretmanager.runtime.config;
+
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import io.quarkus.arc.Arc;
+
+/**
+ * Provides an instance of {@link SecretManagerServiceClient} from the Quarkus application context
+ * with concurrency protection.
+ */
+public class SecretManagerClientProvider {
```

Comment:
Can you make the class package private as it's only accessed from `SecretManagerConfigSource ` ?

Reply:
Done.